src/WpfMath/UserDefinedColorParser.cs

```diff
@@ -0,0 +1,235 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Windows;
+using System.Windows.Media;
+using WpfMath.Exceptions;
+
+namespace WpfMath.Parsers
+{
+    /// <summary>
+    /// Parses definitions of colors from a <see cref="string"/>.
+    /// </summary>
+    public class UserDefinedColorParser
+    {
+        /// <summary>
+        /// Parses the <paramref name="input"/> to its <see cref="Color"/> equivalent.
+        /// </summary>
+        /// <param name="input"></param>
+        /// <returns></returns>
+        public static Color ParseUserColor(string input)
```

Comment:
I recommend return the [`Result` structure](https://github.com/ForNeVeR/wpf-math/blob/b05b6acee7d0d6fa476469b4b2a41a2c5bb8d4f1/src/WpfMath/Utils/Result.cs) here instead of relying on exceptions (we could discuss although!)